### ISSUE TITLE
no way to detect errors and documentation issue [rt.cpan.org #90091]

### ISSUE THREAD

**Issue by toddr** [Perl/perl5#17460]
Migrated from [rt.cpan.org#90091](https://rt.cpan.org/Ticket/Display.html?id=90091) (status was 'new')

Requestors:
* victor@vsespb.ru

From victor@vsespb.ru on 2013-11-06 14:29:14:
```
Hello.

Correct me if I wrong.

> select ( READ, WRITE, EXCEPTION [, TIMEOUT ] )
> ...
> " Upon error an empty list is returned."

however, empty list will be returned when core select() return 0 or -1. in both cases.

And I  think "-1" is "error" code (i.e. errno $! is set), while "0" is not error (one should not test errno).

So, documentation is wrong.

Also there is no way to check for errors in other methods, like can_read etc.

It would be pretty useful to handle errors, like EINTR, and distinct it from other conditions (like timeout and (not sure) eof in filehandles).

```

**Comment by tonycoz** [comment]
The method to distinguish errors from timeout was documented in bc3331071e1ccbdab715397718bb00282f5c5e88.

**Comment by toddr** [comment]
@vsespb I think this report is yours. We're closing it as it seems your concern was already addressed in bc3331071e1ccbdab715397718bb00282f5c5e88. Please let us know if you have any further concerns.
